packages/data-explorer-ui/src/components/Filter/components/VariableSizeList/VariableSizeList.tsx

```diff
@@ -0,0 +1,97 @@
+import { List as MList } from "@mui/material";
+import React, { useCallback, useRef } from "react";
+import {
+  ListChildComponentProps,
+  VariableSizeList as List,
+  VariableSizeListProps as ListProps,
+} from "react-window";
+import {
+  CategoryKey,
+  SelectCategoryValueView,
+} from "../../../../common/entities";
+import { OnFilterFn } from "../../../../hooks/useCategoryFilter";
+import {
+  LIST_ITEM_HEIGHT,
+  MAX_DISPLAYABLE_LIST_ITEMS,
+  MAX_LIST_HEIGHT_PX,
+} from "../../common/constants";
+import VariableSizeListItem from "../VariableSizeListItem/variableSizeListItem";
+
+export type ItemSizeByItemKey = Map<string, number>;
+
+export interface VariableSizeListProps {
+  categoryKey: CategoryKey;
+  height?: ListProps["height"]; // Height of list.
```

Comment:
I've added some additional react-window `VariableSizeList` props i.e. height, width for potential reuse in the filter overall search menu (in case we need them);